ndm-daemonset.Dockerfile

```diff
@@ -0,0 +1,56 @@
+# Copyright 2019-2020 The OpenEBS Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This Dockerfile builds node-disk-manager
+#
+FROM --platform=$BUILDPLATFORM ubuntu as build
+
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
+
+ENV GO111MODULE=off \
+  CGO_ENABLED=0 \
+  DEBIAN_FRONTEND=noninteractive \
+  PATH="/root/go/bin:${PATH}"
+
+WORKDIR /root/go/src/github.com/openebs/node-disk-manager/
+
+RUN apt-get update && apt-get install -y golang make git
+
+COPY . .
+
+RUN export GOOS=$(echo ${TARGETPLATFORM} | cut -d / -f1) && \
+  export GOARCH=$(echo ${TARGETPLATFORM} | cut -d / -f2) && \
+  GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3 | cut -c2-) && \
```

Comment:
question: Is this stage of the build used for building the binary and image in a container?

Reply:
No, this is purely for building the binary, the final stage will be the image we then use. This isn't quite working for myself either.